src/api/ledger/utils.js

```diff
@@ -105,6 +105,25 @@ function hasCompleteLedgerRange(remote: Remote, minLedgerVersion?: number,
     maxLedgerVersion || remote.getLedgerSequence());
 }
 
+function hasLedger(remote: Remote, ledger: number) {
+  return remote.getServer().hasLedger(ledger);
+}
+
+function isNonContiguousLedgerRange(remote: Remote, minLedgerVersion?: number,
+                                    maxLedgerVersion?: number
+): boolean {
+  if (minLedgerVersion === undefined || maxLedgerVersion === undefined) {
+    return false;
+  }
+  return !hasCompleteLedgerRange(remote, minLedgerVersion, maxLedgerVersion)
+    && hasLedger(minLedgerVersion) && hasLedger(maxLedgerVersion);
+}
+
+function isPendingLedgerVersion(remote: Remote, maxLedgerVersion?: number
+): boolean {
+  return remote.getServer().getLastLedger < maxLedgerVersion || 0;
```

Comment:
Should probably use remote.getLedgerSequence(). Also, parens could be good: `(maxLedgerVersion || 0)`.